### ISSUE TITLE
Incorrect codegen for associative table

### ISSUE THREAD

**Issue by acsellers** [stephenafamo/bob#50]
I have a Postgres schema with the following tables:

* users table (id serial, name string, ...)
* roles table (id serial, name string, ...)
* users_roles (role_id, user_id) which are foreign keys pointing at roles.id and users.id

bob generates the AttachRoles/AttachUsers function like the first code snippet. The problem is the for i, rel := range related loop causes an index panic. At the moment, I'm manually editing the for loop to look like the second code snippet.

```
func (o *Role) AttachUsers(ctx context.Context, exec bob.Executor, related ...*User) error {

	var err error
	var usersRoles []*UsersRoleSetter

	for i, rel := range related {
		usersRole := usersRoles[i]
		usersRole.RoleID = omit.From(o.ID)
		usersRole.UserID = omit.From(rel.ID)
	}

	_, err = UsersRolesTable.InsertMany(ctx, exec, usersRoles...)
	if err != nil {
		return fmt.Errorf("inserting related objects: %w", err)
	}

	o.R.Users = append(o.R.Users, related...)

	for _, rel := range related {
		rel.R.Roles = append(rel.R.Roles, o)

	}

	return nil
}
```
```
	for _, rel := range related {
		usersRole := UsersRoleSetter{}
		usersRole.RoleID = omit.From(o.ID)
		usersRole.UserID = omit.From(rel.ID)
		usersRoles = append(usersRoles, &usersRole)
	}
```

**Comment by stephenafamo** [comment]
Thanks for reporting this. It should be fixed in 7879199

If you can confirm that it now works for you, I'll tag and release `v0.20.5`

**Comment by stephenafamo** [comment]
I've released `v0.20.5`
